--- src/fs.c.orig	2010-04-07 00:35:09.953477418 +0200
+++ src/fs.c	2010-04-07 00:36:02.049303958 +0200
@@ -19,31 +19,31 @@
  * to command line options. It is anyway better not to use it...
  */
 static struct FS supported_fs[] = {
-    { "udf", "nosuid,nodev,user", 1, "000", ",iocharset=%s" },
-    { "iso9660", "nosuid,nodev,user", 1, NULL, ",iocharset=%s" },
-    { "vfat", "nosuid,nodev,user,quiet,shortname=mixed", 1, "077", 
+    { "udf", "flush,nosuid,nodev,user", 1, "000", ",iocharset=%s" },
+    { "iso9660", "flush,nosuid,nodev,user", 1, NULL, ",iocharset=%s" },
+    { "vfat", "flush,nosuid,nodev,user,quiet,shortname=mixed", 1, "077", 
       ",iocharset=%s",",fmask=%04o,dmask=%04o"},
-    { "hfsplus", "nosuid,nodev,user", 1, NULL, 0 },
-    { "hfs", "nosuid,nodev,user", 1, "077", NULL, 
+    { "hfsplus", "flush,nosuid,nodev,user", 1, NULL, 0 },
+    { "hfs", "flush,nosuid,nodev,user", 1, "077", NULL, 
       ",file_umask=%04o,dir_umask=%04o"},
-    { "ext4", "nodev,noauto,nosuid,user,errors=remount-ro", 0, NULL, 0 },
-    { "ext3", "nodev,noauto,nosuid,user,errors=remount-ro", 0, NULL, 0 },
-    { "ext2", "nodev,noauto,nosuid,user,errors=remount-ro", 0, NULL, 0 },
-    { "reiserfs", "nodev,noauto,nosuid,user", 0, NULL, 0 },
-    { "reiser4", "nodev,noauto,nosuid,user", 0, NULL, 0 },
-    { "xfs", "nodev,noauto,nosuid,user", 0, NULL, 0 },
-    { "jfs", "nodev,noauto,nosuid,user,errors=remount-ro", 0, NULL, 
+    { "ext4", "flush,nodev,noauto,nosuid,user,errors=remount-ro", 0, NULL, 0 },
+    { "ext3", "flush,nodev,noauto,nosuid,user,errors=remount-ro", 0, NULL, 0 },
+    { "ext2", "flush,nodev,noauto,nosuid,user,errors=remount-ro", 0, NULL, 0 },
+    { "reiserfs", "flush,nodev,noauto,nosuid,user", 0, NULL, 0 },
+    { "reiser4", "flush,nodev,noauto,nosuid,user", 0, NULL, 0 },
+    { "xfs", "flush,nodev,noauto,nosuid,user", 0, NULL, 0 },
+    { "jfs", "flush,nodev,noauto,nosuid,user,errors=remount-ro", 0, NULL, 
       ",iocharset=%s"},
-    { "omfs", "nodev,noauto,nosuid,user", 0, NULL, 0 },
-    { "ntfs-fuse", "nosuid,nodev,user", 1, "077", 
+    { "omfs", "flush,nodev,noauto,nosuid,user", 0, NULL, 0 },
+    { "ntfs-fuse", "flush,nosuid,nodev,user", 1, "077", 
       NULL /* no nls support, it seems*/, ",fmask=%04o,dmask=%04o", 
       1 /* skip detection */
     },
-    { "ntfs-3g", "nosuid,nodev,user", 1, "077", 
+    { "ntfs-3g", "flush,nosuid,nodev,user", 1, "077", 
       NULL /* no nls support, it seems*/, ",fmask=%04o,dmask=%04o", 
       1 /* skip detection */
     },
-    { "ntfs", "nosuid,nodev,user", 1, "077", ",nls=%s" },
+    { "ntfs", "flush,nosuid,nodev,user", 1, "077", ",nls=%s" },
     { NULL, NULL, 0, NULL, 0}
 };
 
