From 563b00f1ac39721bc8192369afddd8b92eeee1fe Mon Sep 17 00:00:00 2001
From: Bartlomiej Zimon <uzi18@o2.pl>
Date: Tue, 12 Jan 2010 19:26:37 +0100
Subject: [PATCH] - gg8,9,10 uses name attribute for image

---
 branches/1.9/src/libgadu.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/branches/1.9/src/libgadu.c b/branches/1.9/src/libgadu.c
index ac8cd43..ad3c3c3 100644
--- a/branches/1.9/src/libgadu.c
+++ b/branches/1.9/src/libgadu.c
@@ -1309,8 +1309,8 @@ static int gg_convert_to_html(char *dst, const char *utf_msg, const unsigned cha
 {
 	const char span_fmt[] = "<span style=\"color:#%02x%02x%02x; font-family:'MS Shell Dlg 2'; font-size:9pt; \">";
 	const int span_len = 75;
-	const char img_fmt[] = "<img src=\"%02x%02x%02x%02x%02x%02x%02x%02x\">";
-	const int img_len = 28;
+	const char img_fmt[] = "<img name=\"%02x%02x%02x%02x%02x%02x%02x%02x\">";
+	const int img_len = 29;
 	int char_pos = 0;
 	int format_idx = 3;
 	unsigned char old_attr = 0;
-- 
1.6.1

