diff --git a/src/torrent/tracker.cc b/src/torrent/tracker.cc
index 76480d7..b778d80 100644
--- a/src/torrent/tracker.cc
+++ b/src/torrent/tracker.cc
@@ -46,7 +46,7 @@ Tracker::Tracker(TrackerList* parent, const std::string& url) :
   m_group(0),
   m_url(url),
 
-  m_normalInterval(1800),
+  m_normalInterval(600),
   m_minInterval(0),
 
   m_scrapeTimeLast(0),
diff --git a/src/torrent/tracker.h b/src/torrent/tracker.h
index 788c1f1..7f209e7 100644
--- a/src/torrent/tracker.h
+++ b/src/torrent/tracker.h
@@ -97,7 +97,7 @@ protected:
 
   void                set_group(uint32_t v)                 { m_group = v; }
 
-  void                set_normal_interval(int v)            { if (v >= 60 && v <= 3600) m_normalInterval = v; }
+  void                set_normal_interval(int v)            { if (v >= 60 && v <= 600) m_normalInterval = v; }
   void                set_min_interval(int v)               { if (v >= 0 && v <= 600)   m_minInterval = v; }
 
   bool                m_enabled;
