diff --git a/src/download/download_wrapper.cc b/src/download/download_wrapper.cc
index 46e194d..f4ef77f 100644
--- a/src/download/download_wrapper.cc
+++ b/src/download/download_wrapper.cc
@@ -268,8 +268,8 @@ DownloadWrapper::receive_tick(uint32_t ticks) {
   if (!info()->is_open())
     return;
 
-  // Every 2 minutes.
-  if (ticks % 4 == 0) {
+  // Every 30s.
+  if (ticks % 1 == 0) {
     if (info()->is_active()) {
       if (info()->is_pex_enabled()) {
         m_main.do_peer_exchange();
