diff -Naur fontconfig-2.8.0.orig/conf.d/10-antialias.conf fontconfig-2.8.0/conf.d/10-antialias.conf
--- fontconfig-2.8.0.orig/conf.d/10-antialias.conf	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.8.0/conf.d/10-antialias.conf	2010-02-20 15:01:02.000000000 +0100
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
+<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<fontconfig>
+<!--  Use the Antialiasing --> 
+  <match target="font">
+    <edit name="antialias" mode="assign"><bool>true</bool></edit>
+  </match>
+</fontconfig>
diff -Naur fontconfig-2.8.0.orig/conf.d/10-hinting.conf fontconfig-2.8.0/conf.d/10-hinting.conf
--- fontconfig-2.8.0.orig/conf.d/10-hinting.conf	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.8.0/conf.d/10-hinting.conf	2010-02-20 15:01:02.000000000 +0100
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
+<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<fontconfig>
+<!--  Use Hinting --> 
+  <match target="font">
+    <edit name="hinting" mode="assign"><bool>true</bool></edit>
+  </match>
+</fontconfig>
diff -Naur fontconfig-2.8.0.orig/conf.d/10-hinting-full.conf fontconfig-2.8.0/conf.d/10-hinting-full.conf
--- fontconfig-2.8.0.orig/conf.d/10-hinting-full.conf	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.8.0/conf.d/10-hinting-full.conf	2010-02-20 15:01:02.000000000 +0100
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
+<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<fontconfig>
+<!--  Use Full Hinting --> 
+  <match target="font">
+    <edit name="hintstyle" mode="assign"><const>hintfull</const></edit>
+  </match>
+</fontconfig>
diff -Naur fontconfig-2.8.0.orig/conf.d/10-hinting-medium.conf fontconfig-2.8.0/conf.d/10-hinting-medium.conf
--- fontconfig-2.8.0.orig/conf.d/10-hinting-medium.conf	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.8.0/conf.d/10-hinting-medium.conf	2010-02-20 15:01:02.000000000 +0100
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
+<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<fontconfig>
+<!--  Use Medium Hinting --> 
+  <match target="font">
+    <edit name="hintstyle" mode="assign"><const>hintmedium</const></edit>
+  </match>
+</fontconfig>
diff -Naur fontconfig-2.8.0.orig/conf.d/10-hinting-slight.conf fontconfig-2.8.0/conf.d/10-hinting-slight.conf
--- fontconfig-2.8.0.orig/conf.d/10-hinting-slight.conf	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.8.0/conf.d/10-hinting-slight.conf	2010-02-20 15:01:02.000000000 +0100
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
+<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<fontconfig>
+<!--  Use Slight Hinting --> 
+  <match target="font">
+    <edit name="hintstyle" mode="assign"><const>hintslight</const></edit>
+  </match>
+</fontconfig>
diff -Naur fontconfig-2.8.0.orig/conf.d/11-lcd-filter-lcddefault.conf fontconfig-2.8.0/conf.d/11-lcd-filter-lcddefault.conf
--- fontconfig-2.8.0.orig/conf.d/11-lcd-filter-lcddefault.conf	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.8.0/conf.d/11-lcd-filter-lcddefault.conf	2010-02-20 15:01:02.000000000 +0100
@@ -0,0 +1,11 @@
+<?xml version="1.0"?>
+<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!-- conf.d/lcd-filter-lcddefault.conf -->
+<fontconfig>
+<!--  Use lcddefault as default for LCD filter -->
+  <match target="font">
+    <edit mode="assign" name="lcdfilter">
+      <const>lcddefault</const>
+    </edit>
+  </match>
+</fontconfig>
diff -Naur fontconfig-2.8.0.orig/conf.d/40-nonlatin.conf fontconfig-2.8.0/conf.d/40-nonlatin.conf
--- fontconfig-2.8.0.orig/conf.d/40-nonlatin.conf	2010-02-20 15:00:37.000000000 +0100
+++ fontconfig-2.8.0/conf.d/40-nonlatin.conf	2010-02-20 15:01:02.000000000 +0100
@@ -51,6 +51,7 @@
 		<family>Baekmuk Dotum</family>
 		<family>SimSun</family>
 		<family>MgOpen Modata</family>
+		<family>MgOpen Moderna</family>
 		<family>Sazanami Gothic</family>
 		<family>AR PL ShanHeiSun Uni</family>
 		<family>ZYSong18030</family>
diff -Naur fontconfig-2.8.0.orig/conf.d/53-monospace-lcd-filter.conf fontconfig-2.8.0/conf.d/53-monospace-lcd-filter.conf
--- fontconfig-2.8.0.orig/conf.d/53-monospace-lcd-filter.conf	1970-01-01 01:00:00.000000000 +0100
+++ fontconfig-2.8.0/conf.d/53-monospace-lcd-filter.conf	2010-02-20 15:01:02.000000000 +0100
@@ -0,0 +1,22 @@
+<?xml version="1.0"?>
+<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!-- conf.d/monospace-lcd-filter.conf -->
+<fontconfig>
+<!--  Use legacy LCD filter on smaller Monospace fonts -->
+  <match target="font">
+    <test name="family">
+      <string>DejaVu Sans Mono</string>
+      <string>Bitstream Vera Sans Mono</string>
+    </test>
+    <test name="pixelsize" compare="less_eq">
+      <double>12.0</double>
+    </test>
+
+    <edit name="lcd_filter" mode="assign">
+      <const>lcdlegacy</const>
+    </edit>
+    <edit name="hintstyle" mode="assign">
+      <const>hintfull</const>
+    </edit>
+  </match>
+</fontconfig>
diff -Naur fontconfig-2.8.0.orig/conf.d/Makefile.am fontconfig-2.8.0/conf.d/Makefile.am
--- fontconfig-2.8.0.orig/conf.d/Makefile.am	2010-02-20 15:00:37.000000000 +0100
+++ fontconfig-2.8.0/conf.d/Makefile.am	2010-02-20 15:01:02.000000000 +0100
@@ -25,13 +25,19 @@
 	README
 
 CONF_FILES = \
+	10-antialias.conf \
 	10-autohint.conf \
+	10-hinting.conf \
+	10-hinting-full.conf \
+	10-hinting-medium.conf \
+	10-hinting-slight.conf \
 	10-no-sub-pixel.conf \
 	10-sub-pixel-bgr.conf \
 	10-sub-pixel-rgb.conf \
 	10-sub-pixel-vbgr.conf \
 	10-sub-pixel-vrgb.conf \
 	10-unhinted.conf \
+	11-lcd-filter-lcddefault.conf \
 	20-fix-globaladvance.conf \
 	20-unhint-small-vera.conf \
 	25-unhint-nonlatin.conf \
@@ -42,6 +48,7 @@
 	49-sansserif.conf \
 	50-user.conf \
 	51-local.conf \
+	53-monospace-lcd-filter.conf \
 	60-latin.conf \
 	65-fonts-persian.conf \
 	65-khmer.conf \
@@ -53,6 +61,10 @@
 	90-synthetic.conf
 
 CONF_LINKS = \
+	10-antialias.conf \
+	10-hinting.conf \
+	10-hinting-slight.conf \
+	11-lcd-filter-lcddefault.conf \
 	20-fix-globaladvance.conf \
 	20-unhint-small-vera.conf \
 	30-urw-aliases.conf \
@@ -62,10 +74,12 @@
 	49-sansserif.conf \
 	50-user.conf \
 	51-local.conf \
+	53-monospace-lcd-filter.conf \
 	60-latin.conf \
 	65-fonts-persian.conf \
 	65-nonlatin.conf \
 	69-unifont.conf \
+	70-no-bitmaps.conf \
 	80-delicious.conf \
 	90-synthetic.conf
 
diff -Naur fontconfig-2.8.0.orig/src/fcname.c fontconfig-2.8.0/src/fcname.c
--- fontconfig-2.8.0.orig/src/fcname.c	2010-02-20 15:00:37.000000000 +0100
+++ fontconfig-2.8.0/src/fcname.c	2010-02-20 15:01:02.000000000 +0100
@@ -454,6 +454,7 @@
     { (FcChar8 *) "lcddefault",	    "lcdfilter",    FC_LCD_DEFAULT },
     { (FcChar8 *) "lcdlight",	    "lcdfilter",    FC_LCD_LIGHT },
     { (FcChar8 *) "lcdlegacy",	    "lcdfilter",    FC_LCD_LEGACY },
+    { (FcChar8 *) "lcdfilterlegacy",	    "lcdfilter",    FC_LCD_LEGACY },
 };
 
 #define NUM_FC_CONSTANTS   (sizeof _FcBaseConstants/sizeof _FcBaseConstants[0])
